packages/ember-glimmer/tests/integration/components/append-test.js

```diff
@@ -35,13 +35,235 @@ class AbstractAppendTest extends RenderingTest {
     this.ids.push(component.elementId);
   }
 
+  ['@test such lifecycle'](assert) {
+    let hooks = [];
+
+    let oldRegisterComponent = this.registerComponent;
+    let componentsByName = {};
+
+    // TODO: refactor/combine with other life-cycle tests
+    this.registerComponent = function(name, _options) {
```

Comment:
Ya, seems fine for now, but we should share this if we can in a future cleanup (maybe make an issue?)...